fluster/test_suite.py

```diff
@@ -332,61 +332,44 @@ def download(
-                print(f"\nError downloading -> {err}\n")
-                pool.terminate()
+        if (
+            len(unique_sources) == 1
+            and len(self.test_vectors) > 1
+            and utils.is_extractable(os.path.basename(next(iter(unique_sources))))
+        ):
+            print(f"Downloading test suite {self.name} using 1 job (single archive)")
+            self._download_single_archive(self.test_vectors, out_dir, self.name, verify, keep_file, retries)
+        else:
+            # Individual downloads (parallel or single)
+            if len(unique_sources) == 1 and len(self.test_vectors) == 1:
+                print(f"Downloading test suite {self.name} using 1 job (single file)")
+                first_tv = next(iter(self.test_vectors.values()))
+                self._download_single(first_tv, out_dir, self.name, verify, extract_all, keep_file, retries)
+            else:
+                print(f"Downloading test suite {self.name} using {jobs} parallel jobs")
+            with Pool(jobs) as pool:
 
-            downloads = []
+                def _callback_error(err: Any) -> None:
+                    print(f"\nError downloading -> {err}\n")
+                    pool.terminate()
 
-            if not self.is_single_archive:
-                print(f"Downloading test suite {self.name} using {jobs} parallel jobs")
-                for test_vector in self.test_vectors.values():
-                    dwork = DownloadWork(
-                        out_dir,
-                        verify,
-                        extract_all,
-                        keep_file,
-                        self.name,
-                        retries,
-                    )
-                    dwork.set_test_vector(test_vector)
-                    downloads.append(
-                        pool.apply_async(
-                            self._download_worker,
-                            args=(dwork,),
-                            error_callback=_callback_error,
-                        )
-                    )
-            else:
-                print(f"Downloading test suite {self.name} using 1 job (no parallel execution possible)")
-                dwork_single_archive = DownloadWorkSingleArchive(
-                    out_dir,
-                    verify,
-                    extract_all,
-                    keep_file,
-                    self.name,
-                    self.test_vectors,
-                    retries,
-                )
-                # We can only use 1 parallel job because all test vectors are inside the same .zip source file
-                downloads.append(
+                downloads = [
                     pool.apply_async(
-                        self._download_worker_single_archive,
-                        args=(dwork_single_archive,),
+                        TestSuite._download_single,
+                        args=(tv, out_dir, self.name, verify, extract_all, keep_file, retries),
```

Comment:
here, if possible use an instance of `DownloadWork` instead of a list of parameters, it helps to keep everything tidy and in context